rfd/0089-merge-webapps.md

```diff
@@ -0,0 +1,185 @@
+yarn binaries. The locations of the enterprise assets will have to be changed
+but that should remain transparent to the day to day activities.
+
+To work on the enterprise version of the application the developer will have to
+check out the enterprise version of Teleport which will bring with it the
+appropriate webapp assets.
+
+Merging these repositories may cause some disruption when multiple developers
+are working on the same feature across the back and frontend. Using a
+[git worktree](https://git-scm.com/docs/git-worktree) to create a worktree of
+the webapps folder is an elegant way of approaching this issue
+
+### Backports
+
+To avoid having different proceses and build systems depending on the version
+of teleport, all branch versions (v9, v10, v11) will have their respective
+versions of the webapps repository.
+
+### Build process
+
+The Teleport build process will need to be expanded to include nodejs and yarn
+in order to be able to build the webapp assets and include into the final
+teleport binaries.
+
+## Process
+
+Below outlines the process for each repository in the order that the steps need
+to be taken. All repositories will be prepped for the final archival and merging
+but the final switch over steps will not be taken until v11 has been released.
+
+The git histories of each branch will be maintained while merging. [Merging repositories](https://stackoverflow.com/questions/13040958/merge-two-git-repositories-without-breaking-file-history)
+
+### Webapps repository
+
+[repository](https://github.com/gravitational/webapps)
+
+#### Actions
+
+- [ ] Triage issues.
+  - [ ] Close those that are no longer relevant.
+  - [ ] [Transfer remaining issues](https://docs.github.com/en/issues/tracking-your-work-with-issues/transferring-an-issue-to-another-repository) to the [Teleport repo](https://github.com/gravitational/teleport)
+  - [ ] Add the [ui label](https://github.com/gravitational/teleport/labels/ui) to
+        those transferred issues.
+- [ ] Triage PRs.
+  - [ ] Close those that are no longer relevant.
+  - [ ] Merge remaining PRs.
+  - [ ] Apply the necessary backports.
+
+### Webapps.e repository
+
+[repository](https://github.com/gravitational/webapps.e)
+
+#### Actions
+
+- [ ] Triage issues.
+  - [ ] Close those that are no longer relevant.
+  - [ ] [Transfer remaining issues](https://docs.github.com/en/issues/tracking-your-work-with-issues/transferring-an-issue-to-another-repository) to the [Teleport.e repo](https://github.com/gravitational/teleport.e)
+  - [ ] Add the [ui label](https://github.com/gravitational/teleport.e/labels/ui)
+- [ ] Triage PRs.
+  - [ ] Close those that are no longer relevant.
+  - [ ] Merge remaining PR's.
+  - [ ] Apply the necessary backports.
+
+### Teleport repository
+
+[repository](https://github.com/gravitational/teleport)
+
+#### Actions
+
+- [ ] Remove `/webassets` submodule
+  - This submodule is no longer required as the web UI will be built on demand.
```

Comment:
Why not retain the `/webassets` submodule and have a script that updates it on every UI PR merge? The same behaviour we currently have (where every `webapps` merge updates the `/webassets` in `teleport`), except within the same repo.

Since the majority of devs working in the `teleport` repo don't make any UI changes, it seems like an unnecessary waste of time to have to build the UI on every Teleport build, this also means they wouldn't be required to have node and yarn installed on their machines. Another approach to mitigate this could be to have a specific `make` script that builds the Teleport binaries but doesn't build the UI. Any thoughts?

Reply:
My 2 cents:

I don't like that the source code and the webassets submodule within the same commit/tag can be out of sync.

Dev's won't need to build the UI on _every_ teleport build, they'll only need to build it once to have a local copy of the assets and then any time they want to recompile UI changes.